shuffle-metrics-reporter/shuffle-metrics-reporter-prometheus/src/main/java/com/alibaba/flink/shuffle/metrics/reporter/AbstractPrometheusReporter.java

```diff
@@ -0,0 +1,346 @@
+        if (metric instanceof Gauge) {
+            ((io.prometheus.client.Gauge) collector)
+                    .setChild(gaugeFrom((Gauge) metric), toArray(dimensionValues));
+        } else if (metric instanceof Counter) {
+            ((io.prometheus.client.Gauge) collector)
+                    .setChild(gaugeFrom((Counter) metric), toArray(dimensionValues));
+        } else if (metric instanceof Meter) {
+            ((io.prometheus.client.Gauge) collector)
+                    .setChild(gaugeFrom((Meter) metric), toArray(dimensionValues));
+        } else if (metric instanceof Histogram) {
+            ((HistogramSummaryProxy) collector).addChild((Histogram) metric, dimensionValues);
+        } else {
+            log.warn(
+                    "Cannot add unknown metric type: {}. This indicates that the metric type is not supported by this reporter.",
+                    metric.getClass().getName());
+        }
+    }
+
+    io.prometheus.client.Gauge.Child gaugeFrom(Gauge gauge) {
+        return new io.prometheus.client.Gauge.Child() {
+            @Override
+            public double get() {
+                final Object value = gauge.getValue();
+                if (value == null) {
+                    log.debug("Gauge {} is null-valued, defaulting to 0.", gauge);
+                    return 0;
+                }
+                if (value instanceof Double) {
+                    return (double) value;
+                }
+                if (value instanceof Number) {
+                    return ((Number) value).doubleValue();
+                }
+                if (value instanceof Boolean) {
+                    return ((Boolean) value) ? 1 : 0;
+                }
+                log.debug(
+                        "Invalid type for Gauge {}: {}, only number types and booleans are supported by this reporter.",
+                        gauge,
+                        value.getClass().getName());
+                return 0;
+            }
+        };
+    }
+
+    private static io.prometheus.client.Gauge.Child gaugeFrom(Counter counter) {
+        return new io.prometheus.client.Gauge.Child() {
+            @Override
+            public double get() {
+                return (double) counter.getCount();
+            }
+        };
+    }
+
+    private static io.prometheus.client.Gauge.Child gaugeFrom(Meter meter) {
+        return new io.prometheus.client.Gauge.Child() {
+            @Override
+            public double get() {
+                return meter.getCount();
+            }
+        };
+    }
+
+    private static String filterCharacters(String input) {
+        // https://prometheus.io/docs/instrumenting/writing_exporters/
+        // Only [a-zA-Z0-9:_] are valid in metric names, any other characters should be sanitized to
+        // an underscore.
+        return UNALLOWED_CHAR_PATTERN.matcher(input).replaceAll("_");
+    }
+
+    private String getGroupByMetricName(
```

Comment:
This is a suboptimal implementation. As discussed above, if the metric name already contains the group name, this is not needed.